File: framework/data-jpa/src/test/java/top/bettercode/simpleframework/data/test/resp/CUser.java
package top.bettercode.simpleframework.data.test.resp;

import java.util.List;
import java.util.Objects;
import top.bettercode.lang.util.StringUtil;

/**
 * @author Peter Wu
 */
public class CUser {

  private String firstName;
  private List<String> lastName;

  public CUser() {
  }

  public String getFirstName() {
    return firstName;
  }

  public void setFirstName(String firstName) {
    this.firstName = firstName;
  }

  public List<String> getLastName() {
    return lastName;
  }

  public void setLastName(List<String> lastName) {
    this.lastName = lastName;
  }

  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (!(o instanceof CUser)) {
      return false;
    }
    CUser cUser = (CUser) o;
    return Objects.equals(firstName, cUser.firstName) && Objects.equals(lastName,
        cUser.lastName);
  }

  @Override
  public int hashCode() {
    return Objects.hash(firstName, lastName);
  }

  @Override
  public String toString() {
    return StringUtil.json(this);
  }
}

File: data-jpa/src/test/kotlin/top/bettercode/summer/data/jpa/resp/CUser.kt
package top.bettercode.summer.data.jpa.resp

import top.bettercode.summer.tools.lang.util.StringUtil.json
import java.util.*

/**
 * @author Peter Wu
 */
class CUser {
    var firstName: String? = null
    var lastName: List<String>? = null
    override fun equals(other: Any?): Boolean {
        if (this === other) {
            return true
        }
        if (other !is CUser) {
            return false
        }
        val cUser = other
        return firstName == cUser.firstName && lastName == cUser.lastName
    }

    override fun hashCode(): Int {
        return Objects.hash(firstName, lastName)
    }

    override fun toString(): String {
        return json(this)
    }
}
